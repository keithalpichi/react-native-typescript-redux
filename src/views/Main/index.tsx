import React, { Component } from "react";
import { View, Text, StyleSheet } from "react-native";

const styles = StyleSheet.create({
  view: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  title: {
    fontSize: 28
  },
  emoji: {
    fontSize: 56
  }
})

class Main extends Component {
  public static navigationOptions = {
    title: "Main Page"
  }

  public render(): JSX.Element {
    return (
      <View style={styles.view}>
        <Text style={styles.title}>react-native-typescript-redux</Text>
        <Text style={styles.emoji}>ðŸŒŠ</Text>
        <Text>A quick start boilerplate for React-Native, Typescript, Redux, and React-Navigation Mobile Apps.</Text>
      </View>
    )
  }
}

export default Main
