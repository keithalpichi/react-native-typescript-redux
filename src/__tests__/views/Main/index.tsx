import * as React from 'react';
import Main from '../../../views/Main';
import renderer from 'react-test-renderer';

describe("<Main />: ", () => {
  it("Renders correctly", () => {
    const tree = renderer.create(<Main />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})