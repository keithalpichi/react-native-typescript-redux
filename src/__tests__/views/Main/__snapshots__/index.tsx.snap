// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main />:  Renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 28,
      }
    }
  >
    react-native-typescript-redux
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 56,
      }
    }
  >
    ðŸŒŠ
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    A quick start boilerplate for React-Native, Typescript, Redux, and React-Navigation Mobile Apps.
  </Text>
</View>
`;
